openmls/src/treesync/mod.rs

```diff
@@ -465,9 +410,12 @@ impl TreeSync {
 
     /// Return a vector containing all [`HpkePublicKey`]s for which we should
     /// have the corresponding private keys.
-    pub(crate) fn owned_encryption_keys(&self) -> Vec<EncryptionKey> {
+    pub(crate) fn owned_encryption_keys(
```

Comment:
Should we move all "own" functions out of the tree?

Reply:
We could do that, but that would mean exposing more functions of `TreeSyncDiff` and `TreeSync` access such as `parent` and `leaf`, but also `filtered_direct_path`. You would also probably have to implement this twice, once for the tree and once for the diff. If it's okay with you, I'd like to punt this to the point where we have a `Treelike` trait that allows us to implement this function once for both tree and diff.